core/src/test/java/org/wso2/msf4j/websocket/DeploymentTest.java

```diff
@@ -0,0 +1,129 @@
+/*
+ *  Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package org.wso2.msf4j.websocket;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.msf4j.MicroservicesRunner;
+import org.wso2.msf4j.util.client.websocket.WebSocketClient;
+import org.wso2.msf4j.websocket.endpoint.ChatAppEndpoint;
+import org.wso2.msf4j.websocket.endpoint.EchoEndpoint;
+import org.wso2.msf4j.websocket.exception.WebSocketEndpointAnnotationException;
+
+import java.net.URISyntaxException;
+import java.nio.ByteBuffer;
+import javax.net.ssl.SSLException;
+
+/**
+ * Test the WebSocket endpoint deployment.
+ * Check all the methods of the endpoint.
+ */
+public class DeploymentTest {
+
+    private static final Logger log = LoggerFactory.getLogger(DeploymentTest.class);
+    private final String host = "localhost";
+    private final String port = "8080";
+    private final int sleepTime = 100;
+
+    private String echoUrl = "ws://" + host + ":" + port + "/echo";
+    private String chatUrl = "ws://" + host + ":" + port + "/chat/";
+
+    private MicroservicesRunner microservicesRunner = new MicroservicesRunner();
+
+    @BeforeClass
+    public void setup() throws WebSocketEndpointAnnotationException {
+        log.info(System.lineSeparator() +
+                         "--------------------------------WebSocket Deployment Test--------------------------------");
```

Comment:
If the test name needs to be highlighted in the console output better to use a fixed format in all tests. Otherwise better not to add long dash lines in individual tests as they might not be consistent.